### ISSUE TITLE
`SheetMetal AddBase` allow `PartDesign::SubShapeBinder` as source

### ISSUE THREAD

**Issue by s-light** [shaise/FreeCAD_SheetMetal#204]
currently [SheetMetal AddBase](https://wiki.freecadweb.org/SheetMetal_AddBase) does not accept a 'PartDesign::SubShapeBinder' as source -
the icon is inactive.

if you manually afterwards change the `.BendSketch` property of the `BaseBend` (`PartDesign::FeaturePython`) Object it is just working. 
so it is only a GUI / pre-selection check that is to limiting.

**Comment by s-light** [comment]
we need to fix the check at 
https://github.com/shaise/FreeCAD_SheetMetal/blob/master/SheetMetalBaseCmd.py#L232

**Comment by s-light** [comment]
fixed by #205

`PartDesign::SubShapeBinder` all working fine.

for `PartDesign::ShapeBinder` this only works as long as the Binder is pointing to the full sketch.
if sub elments are bound it gives an error:
```
07:01:50  Traceback (most recent call last):
  File "/home/stefan/mydata/freecad/.FreeCAD/Mod/sheetmetal/SheetMetalBaseCmd.py", line 144, in execute
    s = smBase(thk = fp.thickness.Value, length = fp.length.Value, radius = fp.radius.Value, Side = fp.BendSide,
  File "/home/stefan/mydata/freecad/.FreeCAD/Mod/sheetmetal/SheetMetalBaseCmd.py", line 77, in smBase
    WireList = MainObject.Shape.Wires[0]
<class 'IndexError'>: list index out of range
``` 
the problem seems  that the `PartDesign::ShapeBinder` does not re-enumerate the edges.
so if edge 0 is not availabe in the list than that fails...
`PartDesign::SubShapeBinder` seems to re-enumerate the edges - so this is is working..

**Comment by s-light** [comment]
example of this behavior in 
[allow_shapeBinder_as_source__missingE0.zip](https://github.com/shaise/FreeCAD_SheetMetal/files/6757864/allow_shapeBinder_as_source__missingE0.zip)
